### PR TITLE
refactor(app-layout): remove ExternalLink component [KHCP-5927]

### PR DESCRIPTION
**Pull request by portikM**
# Summary

Ticket: https://konghq.atlassian.net/browse/KHCP-5927

## PR Checklist

* [x] **Naming & Structure:** the files and package structure use the conventions outlined in the [Creating a Package docs](https://github.com/Kong/public-ui-components/blob/main/docs/creating-a-package.md).
* [x] **Tests pass:** check the output of all package unit and/or component tests.
  * [ ] If this PR is the result of a bug, test coverage was added accordingly.
* [x] **Functional:** all changes do not break existing APIs, but if so, a BREAKING CHANGE commit is in place to bump the major version.
* [x] **Conventional Commits** all commits follow the conventional commit standards [outlined in the main README](https://github.com/Kong/public-ui-components#committing-changes).
* [ ] **Docs:** includes a technically accurate README, and the docs have been updated accordingly based on the changes in this PR.

### CODE REVIEW

**Review by adamdehaven — COMMENTED** [review]
This looks good, but the usage in `shared-ui-components` needs to be updated before this can be merged.

**Comment by portikM** [comment]
> This looks good, but the usage in `shared-ui-components` needs to be updated before this can be merged.

Yup. Added a **Do not merge** label so it isn't merged accidentally.

**Review by adamdehaven — APPROVED** [review]

**Pull request merged**
